patches/minecraft/net/minecraft/world/level/block/RedStoneWireBlock.java.patch

```diff
@@ -0,0 +1,39 @@
+--- a/net/minecraft/world/level/block/RedStoneWireBlock.java
++++ b/net/minecraft/world/level/block/RedStoneWireBlock.java
+@@ -199,7 +_,8 @@
+       BlockState blockstate = p_55523_.m_8055_(blockpos);
+       if (p_55526_) {
+          boolean flag = this.m_55612_(p_55523_, blockpos, blockstate);
+-         if (flag && m_55640_(p_55523_.m_8055_(blockpos.m_7494_()))) {
++         flag &= p_55523_.m_8055_(blockpos.m_7494_()).canRedstoneConnectTo(p_55523_, blockpos.m_7494_(), null);
```

Comment:
This is using a bit wise operator, meaning `canRedstoneConnectTo` will always be run regardless of if `canSurviveOn` returns false. This should be amended to be `flag = flag &&` to use the proper boolean AND operator, which means the jvm will not execute the second half if the first half is already false.

Reply:
Fixed in the latest commit.